src/platform/testing/unit/mocha-setup.js

```diff
@@ -12,15 +12,20 @@ import { JSDOM } from 'jsdom';
 import '../../site-wide/moment-setup';
 import ENVIRONMENTS from 'site/constants/environments';
 
-// import sinon from 'sinon'
-
 global.__BUILDTYPE__ = process.env.BUILDTYPE || ENVIRONMENTS.VAGOVDEV;
 global.__API__ = null;
 global.__MEGAMENU_CONFIG__ = null;
 
 chai.use(chaiAsPromised);
 chai.use(chaiDOM);
 
+function copyProps(src, target) {
+  Object.defineProperties(target, {
+    ...Object.getOwnPropertyDescriptors(src),
+    ...Object.getOwnPropertyDescriptors(target),
+  });
+}
```

Comment:
How is this different from `Object.assign(target, src)`?

Reply:
I'm also curious about this. Did we re-invent a wheel here?

Reply:
This is actually just copy pasta from the [Enzyme example](https://enzymejs.github.io/enzyme/docs/guides/jsdom.html) Eugene linked to. I'm sure there's a distinction too, but I couldn't tell you what it is either. :man_shrugging:

Reply:
The difference between `assign` and `defineProperties` is that this takes "properties", which are objects that include the value of the property as well as sort of "permission bits" in  the form of `writable`, `enumerable`, and `configurable`. JSDom gives `window` some read-only properties, and this is just keeping them at first.

Then you'll notice near the bottom of the setup that I overwrote some of those permissions since there are tests that overwrite the behavior.